src/Microsoft.ML.Transforms/Text/NgramTransform.cs

```diff
@@ -315,14 +315,15 @@ public override void Save(ModelSaveContext ctx)
                 if (_slotNamesTypes[i] != null)
                 {
                     GetSlotNames(i, ref ngramsNames);
-                    Host.Assert(_ngramMaps[i].Count == ngramsNames.Count);
+                    Host.Assert(_ngramMaps[i].Count == ngramsNames.GetValues().Length);
                     Host.Assert(ngramsNames.IsDense);
                     ctx.SaveTextStream(string.Format("{0}-ngrams.txt", Infos[i].Name),
                         writer =>
                         {
-                            writer.WriteLine("# Number of Ngrams terms = {0}", ngramsNames.Count);
-                            for (int j = 0; j < ngramsNames.Count; j++)
-                                writer.WriteLine("{0}\t{1}", j, ngramsNames.Values[j]);
+                            var ngramNameValues = ngramsNames.GetValues();
```

Comment:
>ngramNameValues [](start = 32, length = 15)

`explicitNgramNames`?

Reply:
fixed.

---
In reply to: [233680569](https://github.com/dotnet/machinelearning/pull/1580#discussion_r233680569) [](ancestors = 233680569)